src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3708,267 +3443,146 @@ void PredictSystemLoads(EnergyPlusData &state,
-                AIRRAT(ZoneNum) = AirCap;
-                TempHistoryTerm = AirCap * (3.0 * state.dataHeatBalFanSys->ZTM1(ZoneNum) - (3.0 / 2.0) * state.dataHeatBalFanSys->ZTM2(ZoneNum) +
-                                            (1.0 / 3.0) * state.dataHeatBalFanSys->ZTM3(ZoneNum));
-                TempDepZnLd(ZoneNum) = (11.0 / 6.0) * AirCap + TempDepCoef;
-                TempIndZnLd(ZoneNum) = TempHistoryTerm + TempIndCoef;
-                if (RoomAirflowNetworkZoneInfo(ZoneNum).Node(RoomAirNode).HasHVACAssigned)
-                    RAFNFrac = RoomAirflowNetworkZoneInfo(ZoneNum).Node(RoomAirNode).HVAC(1).SupplyFraction;
-            }
-        } else { // other imperfectly mixed room models
-            TempHistoryTerm = AirCap * (3.0 * state.dataHeatBalFanSys->ZTM1(ZoneNum) - (3.0 / 2.0) * state.dataHeatBalFanSys->ZTM2(ZoneNum) +
-                                        (1.0 / 3.0) * state.dataHeatBalFanSys->ZTM3(ZoneNum));
-            TempDepZnLd(ZoneNum) = (11.0 / 6.0) * AirCap + TempDepCoef;
-            TempIndZnLd(ZoneNum) = TempHistoryTerm + TempIndCoef;
-        }
-
-        // Exact solution or Euler method
-        state.dataHVACGlobal->ShortenTimeStepSysRoomAir = false;
-        if (ZoneAirSolutionAlgo != DataHeatBalance::SolutionAlgo::ThirdOrder) {
-            if (ShortenTimeStepSys && TimeStepSys < state.dataGlobal->TimeStepZone) {
-                if (state.dataHVACGlobal->PreviousTimeStep < state.dataGlobal->TimeStepZone) {
-                    ZoneT1(ZoneNum) = state.dataHeatBalFanSys->ZoneTM2(ZoneNum);
-                    state.dataHeatBalFanSys->ZoneW1(ZoneNum) = state.dataHeatBalFanSys->ZoneWM2(ZoneNum);
-                    if (state.dataRoomAirMod->AirModel(ZoneNum).AirModelType == DataRoomAirModel::RoomAirModel::AirflowNetwork) {
-                        for (LoopNode = 1; LoopNode <= RoomAirflowNetworkZoneInfo(ZoneNum).NumOfAirNodes; ++LoopNode) {
-                            RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempT1 =
-                                RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).AirTempTM2;
-                            RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatW1 =
-                                RoomAirflowNetworkZoneInfo(ZoneNum).Node(LoopNode).HumRatWM2;
-                        }
-                    }
+    for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
+        state.dataZoneTempPredictorCorrector->zoneHeatBalance(zoneNum).predictSystemLoad(
+            state, ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep, zoneNum);
+        if (state.dataHeatBal->doSpaceHeatBalance) {
+            for (int spaceNum : state.dataHeatBal->Zone(zoneNum).spaceIndexes) {
+                state.dataZoneTempPredictorCorrector->spaceHeatBalance(spaceNum).predictSystemLoad(
+                    state, ShortenTimeStepSys, UseZoneTimeStepHistory, PriorTimeStep, zoneNum, spaceNum);
```

Comment:
Example of the new pattern to loop over zones calling a method on `zoneHeatBalance` and optionally loop over spaces to call the same method on `spaceHeatBalance`.